### PR TITLE
Add SparsePauliOp converter

### PR DESCRIPTION
**Pull request by lillian542**
Adds a `convert_sparse_pauli_op_to_pl()` function for converting a [Qiskit SparsePauliOp](https://docs.quantum.ibm.com/api/qiskit/qiskit.quantum_info.SparsePauliOp) into a PennyLane operator.

Although the function is not currently exposed to Sphinx, here is what the rendered docstring looks like:

**Overview**
Screenshot 2024-02-22 at 2 57 56 PM

**Usage Details**
Screenshot 2024-02-22 at 2 58 06 PM

### CODE REVIEW

**Comment by Mandrenkov** [comment]
Unfortunately, I can't tag you a reviewer @lillian542 but feel free to leave a review and comment "I approve." or similar when you're done. That will be my cue to formally approve the PR on your behalf.

**Review by lillian542 — COMMENTED** [review]
What if we make the primary example just have floats as coefficients instead of Parameters, and then add a Usage Details section that shows passing parameter values and passing wires? 

Left a few other small doc suggestions as well, see what you think!

But I approve 👍

File: pennylane_qiskit/converter.py [hunk 1]
```diff
@@ -562,6 +563,74 @@ def load_qasm_from_file(file: str):
     return load(QuantumCircuit.from_qasm_file(file))
 
 
+def convert_sparse_pauli_op_to_pl(
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply pennylane_qiskit/converter.py:566]: @lillian542 What do you think about `load_sparse_pauli_op()` or `load_pauli_op()` for consistency?

File: pennylane_qiskit/converter.py [hunk 2]
```diff
@@ -562,6 +563,74 @@ def load_qasm_from_file(file: str):
...
+            to learn more about the expected format of these parameters
+
+    Returns:
+        pennylane.operation.Operator: The equivalent PennyLane operator.
+
+    .. note::
+
+        The order in which Pauli operations appear in each SparsePauliOp term is the reverse of how
+        they are applied in the PennyLane operator. This is consistent with the iteration order of a
+        Pauli term in Qiskit.
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply pennylane_qiskit/converter.py:589]: Thanks, great suggestion! I like the idea of documenting a concrete example.

[the remark this answers is not in the document]
Comment (Mandrenkov) [reply pennylane_qiskit/converter.py:589]: For reference, here is the updated note:

Screenshot 2024-02-22 at 1 30 34 PM

File: pennylane_qiskit/converter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lillian542) [reply pennylane_qiskit/converter.py]: Good idea, I guess let's do `load_pauli_op`?

**Comment by Mandrenkov** [comment]
Thanks for the suggestions, @lillian542! I've updated the docstring and renamed the function accordingly.

File: pennylane_qiskit/converter.py [hunk 3]
```diff
@@ -550,6 +551,121 @@ def load_qasm_from_file(file: str):
...
+            from qiskit.circuit import Parameter
+
+            a, b, c = [Parameter(var) for var in "abc"]
+            param_qiskit_op = SparsePauliOp(["II", "XZ", "YX"], coeffs=np.array([a, b, c]))
+
+        defines a ``SparsePauliOp`` with three coefficients (parameters):
+
+        >>> param_qiskit_op
+        SparsePauliOp(['II', 'XZ', 'YX'],
+                  coeffs=[ParameterExpression(1.0*a), ParameterExpression(1.0*b),
```
Comment (lillian542) [inline pennylane_qiskit/converter.py:620]: looks like the third parameter and close parentheses got cut off here

Comment (Mandrenkov) [reply pennylane_qiskit/converter.py]: Good catch! 🎣  I must have missed this when updating the docstring.

**Review by Mandrenkov — APPROVED** [review]
Approving on behalf of @lillian542.

**Pull request merged**